src/main/java/it/pagopa/pn/paperchannel/middleware/queue/consumer/handler/SendToDeliveryPushHandler.java

```diff
@@ -36,4 +43,19 @@ public Mono<Void> handleMessage(PnDeliveryRequest entity, PaperProgressStatusEve
         return Mono.empty();
     }
 
+    /**
+     * Update delivery request setting refined field to true
+     *
+     * @param pnDeliveryRequest request to update
+     * */
+    private void updateRefinedDeliveryRequest(PnDeliveryRequest pnDeliveryRequest) {
```

Comment:
```java
è voluto il subsribe in modo tale che anche se c'è un errore si continui il flusso possiamo creare una chain del tipo:

public Mono<Void> handleMessage(PnDeliveryRequest entity, PaperProgressStatusEventDto paperRequest) {
        log.debug("[{}] Sending to delivery-push or event-bridge", paperRequest.getRequestId());
        log.debug("[{}] Response of ExternalChannel from request id {}", paperRequest.getRequestId(), paperRequest);
        SendEvent sendEvent = SendEventMapper.createSendEventMessage(entity, paperRequest);

        return updateRefinedEntityIfStatusIsOk(entity)
                .doOnNext(pnDeliveryRequest -> {
                    if (Utility.isCallCenterEvoluto(entity.getRequestId())) {
                        String clientId = MDC.get(Const.CONTEXT_KEY_CLIENT_ID);
                        log.debug("[{}] clientId from context", clientId);
                        sqsSender.pushSendEventOnEventBridge(clientId, sendEvent);
                        log.info("[{}] Sent to event-bridge: {}", paperRequest.getRequestId(), sendEvent);
                    } else {
                        sqsSender.pushSendEvent(sendEvent);
                        log.info("[{}] Sent to delivery-push: {}", paperRequest.getRequestId(), sendEvent);
                    }
                })
                .then();
    }
    
    private Mono<PnDeliveryRequest> updateRefinedEntityIfStatusIsOk(PnDeliveryRequest pnDeliveryRequest) {
        log.debug("[{}] Updating DeliveryRequest with refinement information", pnDeliveryRequest.getRequestId());
        if (StatusCodeEnum.OK.getValue().equals(pnDeliveryRequest.getStatusDetail())) {
            pnDeliveryRequest.setRefined(true);
            return this.requestDeliveryDAO
                    .updateData(pnDeliveryRequest)
                    .doOnError(ex -> log.warn("[{}] Error while setting request as refined", pnDeliveryRequest.getRequestId(), ex));
        }
        else {
            return Mono.just(pnDeliveryRequest);
        }
        
    }
```

Reply:
Mi torna gestirlo in questo modo, per il momento si era preferito usare il subscribe per lasciare i flussi di send in coda e salvataggio completamente scollegati tra loro e non creare la chain.
Però mi torna anche in ottica di gestione futura, lo aggiorno